Mathlib/Data/BitVec/Lemmas.lean

```diff
@@ -218,4 +244,66 @@ instance : CommSemiring (BitVec w) :=
     toFin_zero toFin_one toFin_add toFin_mul (Function.swap toFin_nsmul)
     toFin_pow toFin_natCast
 
+/-!
+## Extensionality
+-/
+
+/-- If two bitvectors agree on all in-bound bits, then they agree on all bits -/
+theorem getLsb_eq_of_getLsb' {x y : BitVec w} (h : ∀ (i : Fin w), x.getLsb' i = y.getLsb' i) :
+    ∀ (i : ℕ), x.getLsb i = y.getLsb i := by
+  simp only [getLsb, testBit]
+  intro i
+  cases' lt_or_le i w with hi hi
+  · exact h ⟨i, hi⟩
+  · have (z : BitVec w) : z.toNat < 2 ^ i :=
+      Nat.lt_of_lt_of_le z.toNat_lt (pow_le_pow_right (le_succ 1) hi)
+    rw [Nat.shiftRight_eq_zero_of_lt (this x), Nat.shiftRight_eq_zero_of_lt (this y)]
+
+/-- If two bitvectors agree on all bits, then they are equal. See also `Std.BitVec.ext_msb` -/
+@[ext]
+theorem ext_lsb {x y : BitVec w} (h : ∀ i, x.getLsb' i = y.getLsb' i) : x = y := by
```

Comment:
I wouldn't put the extensionality attribute on this lemma. I don't think this is the first thing people expect when they think of an extensionality for bitvectors, and the `ext` tactic should usually use the first thing people expect. I don't think anything is the first thing people expect for bitvectors to be honest.

Reply:
This is the first thing I'd expect; "two bitvectors are equal if they have the same bits; but maybe this is like `Complex.ext`, which is the "obvious" ext lemma, but often you don't actually want it.

Reply:
I don't feel that strongly about this comment, I'll happily leave the `ext` attribute there.

Reply:
I would prefer to keep it as `ext`. It seems quite natural for me. I resolve this for now as @ChrisHughes24 does not seem to feel strongly here.